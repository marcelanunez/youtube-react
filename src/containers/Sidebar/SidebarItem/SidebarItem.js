import React from 'react';
import { Icon, Menu, MenuItem,  } from 'semantic-ui-react';
import './SidebarItem.scss'

export const SidebarItem = (props) => {
  const highlight = props.highlight ? 'highlight-item' : null;
  return (
    <Menu.Item>
      <div className={['sidebar-item', highlight].join(' ')}>
        <span>
          <Icon name={props.icon} size='large'/>
        </span>
        <span>
          {props.label}
        </span>
      </div>
    </Menu.Item>
  );
}
export default SidebarItem;