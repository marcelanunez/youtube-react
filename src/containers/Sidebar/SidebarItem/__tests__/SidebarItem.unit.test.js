import React from 'react';
import { shallow } from 'enzyme';
import { SidebarItem } from '../SidebarItem';

describe('SidebarItem', () => {
  test('renders empty SidebarItem', () => {
    const wrapper = shallow(<SidebarItem />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders highlighted SidebarItem that navigates to /feed/trending', () => {
    const wrapper = shallow(
      <SidebarItem highlighted icon="fire" label="Trending" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('render non-highlighted SidebarItem that navigates to /feed/trending', () => {
    const wrapper = shallow(<SidebarItem icon="fire" label="Trending" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Renders highlighted SidebarItem with no navigation', () => {
    const wrapper = shallow(
      <SidebarItem highlighted icon="fire" label="Trending" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
