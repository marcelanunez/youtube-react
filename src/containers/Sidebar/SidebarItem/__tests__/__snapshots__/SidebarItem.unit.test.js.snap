// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarItem Renders highlighted SidebarItem with no navigation 1`] = `
<MenuItem>
  <div
    className="sidebar-item "
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;

exports[`SidebarItem render non-highlighted SidebarItem that navigates to /feed/trending 1`] = `
<MenuItem>
  <div
    className="sidebar-item "
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;

exports[`SidebarItem renders empty SidebarItem 1`] = `
<MenuItem>
  <div
    className="sidebar-item "
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;

exports[`SidebarItem renders highlighted SidebarItem that navigates to /feed/trending 1`] = `
<MenuItem>
  <div
    className="sidebar-item "
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;
